### PR TITLE
[1254] Add guard to prevent rendering on Find

### PR DESCRIPTION
**Pull request by avinhurry**
### Context

The incomplete section inset link was found on Find. This component is for preview only.

### Changes proposed in this pull request

Add a gaurd to prevent the component rendering on Find

### Guidance to review

Check that the component no longer renders in the review app. 

- Course on preview: https://publish-teacher-training-pr-3534.london.cloudapps.digital/publish/organisations/17B/2023/courses/Q902/preview

- Find the course above on Find, following this link and ensure the component does not render: https://find-pr-3534.london.cloudapps.digital

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased main
- [x] Cleaned commit history
- [x] Tested by running locally
- [x] Inform data insights team due to database changes

### CODE REVIEW

**Comment by avinhurry** [comment]
[[Bug] Enter details about the accredited provider](https://trello.com/c/E9Lax3ey/1254-bug-enter-details-about-the-accredited-provider)

File: app/components/course_preview/missing_information_component.rb [hunk 1]
```diff
@@ -21,7 +21,11 @@ def link
     end
 
     def render?
-      FeatureService.enabled?(:course_preview_missing_information)
+      FeatureService.enabled?(:course_preview_missing_information) && preview?
+    end
+
+    def preview?
+      params[:action] == 'preview' || params[:action].nil?
```
Comment (darokel) [inline app/components/course_preview/missing_information_component.rb:28]: i don't think the components should reaching directly into the params, it should probably be passed in via an args and will make testing easier

Comment (avinhurry) [reply app/components/course_preview/missing_information_component.rb:28]: How do you mean?

Comment (darokel) [reply app/components/course_preview/missing_information_component.rb:28]:
```ruby
def initialize(course:, information_type:, preview_mode: false)
      super
      @information_type = information_type
      @course = course
end

MissingInformationComponent.new(course:, information_type: :some_value, preview_mode: params[:action] == 'preview')

# or

MissingInformationComponent.new(course:, information_type: :some_value, action: params[:action])
```

Comment (avinhurry) [reply app/components/course_preview/missing_information_component.rb]: Oh right. Sorry, it was showing me your unedited comment for some reason 🤷🏼‍♂️. Will this do? 

https://github.com/DFE-Digital/publish-teacher-training/pull/3534/commits/c3f48c425a9ad332a85166d51bf23a16978848b6

File: app/components/course_preview/missing_information_component.rb [hunk 2]
```diff
@@ -2,14 +2,15 @@
 
 module CoursePreview
   class MissingInformationComponent < ViewComponent::Base
-    attr_accessor :information_type, :course
+    attr_accessor :information_type, :course, :is_preview
```
Comment (darokel) [inline app/components/course_preview/missing_information_component.rb:5]:
```suggestion
attr_reader :information_type, :course, :is_preview
```

File: app/components/course_preview/missing_information_component.rb [hunk 3]
```diff
@@ -21,7 +22,7 @@ def link
     end
 
     def render?
-      FeatureService.enabled?(:course_preview_missing_information)
+      FeatureService.enabled?(:course_preview_missing_information) && @is_preview
```
Comment (darokel) [inline app/components/course_preview/missing_information_component.rb:25]:
```
FeatureService.enabled?(:course_preview_missing_information) && is_preview
```

**Review by darokel — APPROVED** [review]

**Review by victoriawinlow — APPROVED** [review]

**Pull request merged**